homeassistant/components/nibe_heatpump/climate.py

```diff
@@ -225,3 +226,25 @@ async def async_set_temperature(self, **kwargs: Any) -> None:
 
         if (temperature := kwargs.get(ATTR_TARGET_TEMP_HIGH)) is not None:
             await coordinator.async_write_coil(self._coil_setpoint_cool, temperature)
+
+    async def async_set_hvac_mode(self, hvac_mode: HVACMode) -> None:
+        """Set new target hvac mode."""
+        coordinator = self.coordinator
+
+        if hvac_mode == HVACMode.HEAT_COOL:
+            await coordinator.async_write_coil(
+                self._coil_cooling_with_room_sensor, "ON"
+            )
+            await coordinator.async_write_coil(self._coil_use_room_sensor, "ON")
+        elif hvac_mode == HVACMode.HEAT:
+            await coordinator.async_write_coil(
+                self._coil_cooling_with_room_sensor, "OFF"
+            )
+            await coordinator.async_write_coil(self._coil_use_room_sensor, "ON")
+        elif hvac_mode == HVACMode.AUTO:
+            await coordinator.async_write_coil(
+                self._coil_cooling_with_room_sensor, "OFF"
+            )
+            await coordinator.async_write_coil(self._coil_use_room_sensor, "OFF")
+        else:
+            raise HomeAssistantError(f"Mode not supported {hvac_mode}")
```

Comment:
```suggestion
            raise HomeAssistantError(f"{hvac_mode} mode not supported for {self.name}")
```
Perhaps write it a bit more user-friendly